ompi/mca/coll/ucc/coll_ucc_allreduce.c

```diff
@@ -0,0 +1,97 @@
+
+/**
+ * Copyright (c) 2021 Mellanox Technologies. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ */
+
+#include "coll_ucc_common.h"
+
+static inline ucc_status_t mca_coll_ucc_allreduce_init(const void *sbuf, void *rbuf, int count,
+                                                       struct ompi_datatype_t *dtype,
+                                                       struct ompi_op_t *op, mca_coll_ucc_module_t *ucc_module,
+                                                       ucc_coll_req_h *req,
+                                                       mca_coll_ucc_req_t *coll_req)
+{
+    ucc_datatype_t         ucc_dt;
+    ucc_reduction_op_t     ucc_op;
+
+    ucc_dt = ompi_dtype_to_ucc_dtype(dtype);
+    ucc_op = ompi_op_to_ucc_op(op);
+    if (OPAL_UNLIKELY(COLL_UCC_DT_UNSUPPORTED == ucc_dt ||
+                      COLL_UCC_OP_UNSUPPORTED == ucc_op)) {
+        UCC_VERBOSE(5, "ompi_datatype is not supported: dtype = %s",
```

Comment:
The error message only partially reflect the checks.

Reply:
fixed